patchwork-extensions-bakedmodel/src/main/java/net/patchworkmc/mixin/extensions/bakedmodel/MixinBlockModelRenderer.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2020, 2019-2020
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.patchworkmc.mixin.extensions.bakedmodel;
+
+import java.util.List;
+import java.util.Random;
+
+import org.objectweb.asm.Opcodes;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.At.Shift;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import net.minecraftforge.client.extensions.IForgeBakedModel;
+import net.minecraftforge.client.model.data.IModelData;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.client.render.BufferBuilder;
+import net.minecraft.client.render.block.BlockModelRenderer;
+import net.minecraft.client.render.model.BakedModel;
+import net.minecraft.client.render.model.BakedQuad;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import net.minecraft.world.BlockRenderView;
+
+import net.patchworkmc.impl.extensions.bakedmodel.ModelDataParameter;
+import net.patchworkmc.impl.extensions.bakedmodel.ForgeBlockModelRenderer;
+
+/**
+ * Implements {@link net.minecraftforge.client.extensions.IForgeBakedModel#getQuads(BlockState,Direction,Random,IModelData)} and
+ * {@link net.minecraftforge.client.extensions.IForgeBakedModel#shouldApplyDiffuseLighting()}.
+ */
```

Comment:
This is really messy, but to be honest I don't understand the code here. Unless someone is able to show how it's done otherwise, though, this is the solution I'll accept for now.